chord_metadata_service/chord/workflows/wdls/experiments_json_with_files.wdl

```diff
@@ -146,20 +146,47 @@ task post_to_drs {
         String token
         Boolean validate_ssl
     }
+
     command <<<
+        # Extract project_id and dataset_id
         project_id=$(python3 -c 'print("~{project_dataset}".split(":")[0])')
         dataset_id=$(python3 -c 'print("~{project_dataset}".split(":")[1])')
-        curl ~{true="" false="-k" validate_ssl} \
+
+        resp_main=$(curl ~{true="" false="-k" validate_ssl} \
             -X POST \
             -F "file=@~{file_path}" \
             -F "project_id=$project_id" \
             -F "dataset_id=$dataset_id" \
             -H "Authorization: Bearer ~{token}" \
             --fail-with-body \
-            "~{drs_url}/ingest"
+            "~{drs_url}/ingest")
+        echo "$resp_main" | jq -c
+
+        # If it's BAM or CRAM, ingest the index as well
```

Comment:
this part of the code still isn't matching desired behaviour exactly - indices should be ingested if they're listed in the JSON; or, if no index is specified in the JSON, one should be generated (maybe in the python script using subprocess?) and put into the JSON. I think also `prepare_for_drs` should extract index files as well as file paths, and this task should stay naive to the type of file it is ingesting.

Reply:
I think for now the best way to write this workflow might unfortunately just be as a single big Python task which uses the `requests` library to ingest into DRS and, if needed, uses subprocess+samtools to generate indices for files missing the indices, rewriting the experiments JSON as it goes and then POSTing it to Katsu at the end.